charts/castai-db-optimizer/templates/pgcat-config.yaml

```
@@ -2,18 +2,35 @@
 {{- if not .Values.pooling.databases }}
 {{- fail "pooling.databases cannot be empty when pooling is enabled. Please provide at least one database name." }}
 {{- end }}
+{{- $secretName := "" }}
+{{- if .Values.pooling.credentialsSecretRef }}
+{{- $secretName = .Values.pooling.credentialsSecretRef.name }}
+{{- else }}
+{{- $secretName = printf "%s-pooling-credentials" (include "name" .) }}
+{{- end }}
+{{- $credentials := (lookup "v1" "Secret" .Release.Namespace $secretName) }}
+{{- $username := "" }}
+{{- $password := "" }}
+{{- if $credentials }}
+{{- $username = ($credentials.data.username | b64dec) }}
+{{- $password = ($credentials.data.password | b64dec) }}
```

Because the secret stores these as `USERNAME` and `PASSWORD` (uppercase) for this to work correctly it has to be:

```suggestion
{{- $username = ($credentials.data.USERNAME | b64dec) }}
{{- $password = ($credentials.data.PASSWORD | b64dec) }}
```

Otherwise, you'd get:

> Error: UPGRADE FAILED: template: castai-db-optimizer/templates/pgcat-config.yaml:15:46: executing "castai-db-optimizer/templates/pgcat-config.yaml" at <b64dec>: invalid value; expected string